File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-4.java
class Solution {
    public int findMin(int[] nums) {
        int min = 0;
        int max = nums.length - 1;
        int minV = nums[min];
        while (min < max) {
            int mid = (min + max) / 2;
            if (nums[mid] < nums[max]) { 
                max = mid;
            }
            else min = mid + 1;
        }
        return nums[min];
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        max_length = 0
        num_set = set()
        for n in nums:
            num_set.add(n)

        for n in num_set:
            if n - 1 not in num_set:
                length = 0
                while n + length in num_set:
                    length += 1
                max_length = max(length, max_length)

        return max_length